### ISSUE TITLE
webxdc: `.wasm` files don't work

### ISSUE THREAD

**Issue by WofWca** [deltachat/deltachat-android#2635]
I get this error:

```
Failed to execute 'compile' on 'WebAssembly': Incorrect response MIME type. Expected 'application/wasm'.
```

Responsible code:
https://github.com/deltachat/deltachat-android/blob/c4a44694247ee1082a07d4bacd55b1d26064c175/src/org/thoughtcrime/securesms/WebxdcActivity.java#L237-L246

~~In DC Desktop there is no MIME type "white-list"~~
Here's the relevant Desktop code:
https://github.com/deltachat/deltachat-desktop/blob/5accdc9718ebd93461c38e346daa518e2096364f/src/main/deltachat/webxdc.ts#L145-L192

Test file (rename to `.xdc`): [doom.zip](https://github.com/deltachat/deltachat-android/files/12323360/doom.zip)

Related: https://github.com/webxdc/webxdc-test/pull/15

**Comment by r10s** [comment]
> In DC Desktop there is no MIME type "white-list"

imu, also on android, there is no "White List": it just seems as if `getMimeTypeFromExtension()` does not recognise the correct mime type and we do not correct he error

**Comment by WofWca** [comment]
> imu, also on android, there is no "White List"

Whoops, you're right, I missed the `if` condition.

> for doom.zip: that also does not work in a browser

Huh? How did you test? It works for me, with `python3 -m http.server`.

> what would be the correct type?

The error implies that it's supposed to be `'application/wasm'`, which is the case for `python3 -m http.server`.

**Comment by r10s** [comment]
> Huh? How did you test? It works for me, with python3 -m http.server.

i just opened index.html from the .xdc in a browser - usually, that should work (maybe in this case not as again the mime type is wrong)

**Comment by r10s** [comment]
i created a pr, https://github.com/deltachat/deltachat-android/pull/2636, CI should create and .apk that can be used for testing.

as doom.xdc is too large, please also add a little thing to `webxdc-test.xdc` - the current test does not involve .wasm files

**Comment by r10s** [comment]
closed by #2636 

for the missing test in webxdc-test i created https://github.com/webxdc/webxdc-test/issues/20 and assigned @WofWca - @WofWca  if you cannot add the test, please unassign yourself
